test/e2e/framework/framework.go

```diff
@@ -86,25 +84,48 @@ var (
 	KubeClients []*kubeclientset.Clientset
 )
 
-// NewFramework creates a test framework.
-func NewFramework(baseName string) *Framework {
+// NewBareFramework creates a test framework, without ginkgo dependencies
+func NewBareFramework(baseName string) *Framework {
 	f := &Framework{
 		BaseName:           baseName,
 		namespacesToDelete: map[string]bool{},
 	}
-
-	ginkgo.BeforeEach(f.BeforeEach)
-	ginkgo.AfterEach(f.AfterEach)
-
 	return f
 }
 
 func AddBeforeSuite(beforeSuite func()) {
 	beforeSuiteFuncs = append(beforeSuiteFuncs, beforeSuite)
 }
 
+var By func(string)
+var Fail func(string)
+var userAgentFunction func() string
+
+func SetStatusFunction(by func(string)) {
+	By = by
+}
+func SetFailFunction(fail func(string)) {
+	Fail = fail
+}
+
+func SetUserAgentFunction(uaf func() string) {
+	userAgentFunction = uaf
+}
+
+func init() {
+	By = func(str string) {
+		fmt.Println(str)
+	}
+	Fail = func(str string) {
+		panic("Framework Fail:" + str)
+	}
+	userAgentFunction = func() string {
+		return "shipyard-framework-agent"
+	}
+}
+
 func BeforeSuite() {
-	ginkgo.By("Creating kubernetes clients")
+	By("Creating kubernetes clients")
 
 	if len(TestContext.KubeConfig) > 0 {
 		Expect(len(TestContext.KubeConfigs)).To(BeZero(),
```

Comment:
Do we need a wrap to `Except`?

Reply:
That's a gomega dependency, it's ok, that one is note easily switchable, but framework users can just do:
```
gomega.RegisterFailHandler(func(message string, callerSkip ... int) { panic(message) })
```
For now